plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -68,33 +66,65 @@ public String getDescription() {
      * Unmodifiable list containing each processor that operates on the data.
      */
     public List<Processor> getProcessors() {
-        return processors;
+        return compoundProcessor.getProcessors();
     }
 
     public final static class Factory {
 
+        private Processor readProcessor(Map<String, Processor.Factory> processorRegistry, String type, Map<String, Object> config) throws Exception {
+            Processor.Factory factory = processorRegistry.get(type);
+            if (factory != null) {
+                List<Map<String, Map<String, Object>>> onFailureProcessorConfigs = ConfigurationUtils.readOptionalList(config, "on_failure");
```

Comment:
maybe add a private helper method for parsing on failure processors that both readProcessor(...) and create(...) use?